core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -539,11 +539,21 @@ object ZIOSpec extends ZIOBaseSpec {
     ),
     suite("foreachPar_")(
       testM("accumulates errors") {
-        val failures = ZIO
-          .foreachPar_(1 to 3)(IO.fail(_).uninterruptible)
-          .foldCause(_.failures.toSet, _ => Set.empty)
-        assertM(failures)(equalTo(Set(1, 2, 3)))
-      } @@ flaky,
+        def task(started: Ref[Int], trigger: Promise[Nothing, Unit])(i: Int): IO[Int, Unit] =
```

Comment:
Why do we need to change the test here?

Reply:
Ah, I see, you are making sure that the effects aren't pre-interrupted. Makes sense.

Reply:
Yes, sadly the simple version of the test is still flaky as the fibers can be interrupted before they enter the uninterruptible region.